source/dem/disable_particle_contact.cc

```diff
@@ -0,0 +1,335 @@
+
+                  for (int d = 0; d < dim; ++d)
+                    {
+                      cell_velocity_fluctuation_squared_sum[d] +=
+                        (particle_properties[DEM::PropertiesIndex::v_x + d] -
+                         velocity_cell_average[d]) *
+                        (particle_properties[DEM::PropertiesIndex::v_x + d] -
+                         velocity_cell_average[d]);
+                    }
+                }
+
+              // Calculate average granular temperature in the cell
+              for (int d = 0; d < dim; ++d)
+                {
+                  cell_velocity_fluctuation_squared_average[d] =
+                    cell_velocity_fluctuation_squared_sum[d] /
+                    n_particles_in_cell;
+                  granular_temperature_cell +=
+                    (1.0 / dim) * cell_velocity_fluctuation_squared_average[d];
+                }
+            }
+
+          // Store the average granular temperature and solid fraction with
+          // active cell index
+          granular_temperature_average[cell->active_cell_index()] =
+            granular_temperature_cell;
+          solid_fractions[cell->active_cell_index()] = solid_fraction;
+        }
+    }
+}
+
+
+template <int dim>
+void
+DisableParticleContact<dim>::identify_mobility_status(
+  const DoFHandler<dim> &                background_dh,
+  const Particles::ParticleHandler<dim> &particle_handler,
+  MPI_Comm                               mpi_communicator)
+{
+  // Reset cell status containers
+  status_to_cell.clear();
+  status_to_cell.resize(mobility_status::n_mobility_status);
+
+  // Create dummy dofs for background dof handler
+  const FE_Q<dim>    fe(1);
+  const unsigned int dofs_per_cell = fe.dofs_per_cell;
+
+  // Get locally owned and relevant dofs
+  const IndexSet locally_owned_dofs = background_dh.locally_owned_dofs();
+  IndexSet       locally_relevant_dofs =
+    DoFTools::extract_locally_relevant_dofs(background_dh);
+
+  // Reinit all value of mobility at nodes
+  mobility_at_nodes.reinit(locally_owned_dofs,
+                           locally_relevant_dofs,
+                           mpi_communicator);
+  mobility_at_nodes = 0;
+
+  // Empty status (3) to nodes when no particle in cell
+  for (auto cell = active_ghost_cells.begin();
+       cell != active_ghost_cells.end();)
+    {
+      // Check if the cell has any particles
+      if (particle_handler.n_particles_in_cell(*cell) == 0)
+        {
+          std::vector<types::global_dof_index> local_dofs_indices(
+            dofs_per_cell);
+          (*cell)->get_dof_indices(local_dofs_indices);
+
+          // Remove empty cell from cells to reduce the number of check
+          cell = active_ghost_cells.erase(cell);
+
+          // Assign empty status to nodes
+          for (auto dof_index : local_dofs_indices)
+            {
+              mobility_at_nodes(dof_index) = mobility_status::empty;
+            }
+        }
+      else
+        {
```

Comment:
Are those else necessary?

Reply:
Yes, at line 183, calling .erase() will return the next iterator, but if the particle is not emprty, we have to "manually" iterate on the next cell